core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -173,6 +176,18 @@ public static String getStringOrNull(String property, JsonNode node) {
     return getString(property, node);
   }
 
+  public static ByteBuffer getByteBufferOrNull(String property, JsonNode node) {
+    if (!node.has(property) || node.get(property).isNull()) {
+      return null;
+    }
+
+    JsonNode pNode = node.get(property);
+    Preconditions.checkArgument(
+        pNode.isTextual(), "Cannot parse from non-text value: %s: %s", property, pNode);
```

Comment:
nit: maybe we should mention that we're trying to parse this from text to a binary representation

Reply:
I also fixed a couple other error msgs with the same problem.